diff --git a/src/main/template/param.handlebars b/src/main/template/param.handlebars
index d851632..e20b92c 100644
--- a/src/main/template/param.handlebars
+++ b/src/main/template/param.handlebars
@@ -20,9 +20,9 @@
 			<div class="parameter-content-type" />
 		{{else}}
 			{{#if defaultValue}}
-				<input class='parameter' minlength='0' name='{{name}}' placeholder='' type='text' value='{{defaultValue}}'/>
+				<input data-threescale-name='{{threescale_name}}' class='parameter' minlength='0' name='{{name}}' placeholder='' type='text' value='{{defaultValue}}'/>
 			{{else}}
-				<input class='parameter' minlength='0' name='{{name}}' placeholder='' type='text' value=''/>
+				<input data-threescale-name='{{threescale_name}}' class='parameter' minlength='0' name='{{name}}' placeholder='' type='text' value=''/>
 			{{/if}}
 		{{/if}}
 	{{/if}}
diff --git a/src/main/template/param_required.handlebars b/src/main/template/param_required.handlebars
index 69e9370..231034a 100644
--- a/src/main/template/param_required.handlebars
+++ b/src/main/template/param_required.handlebars
@@ -17,9 +17,9 @@
 			<input class='parameter' class='required' type='file' name='{{name}}'/>
 		{{else}}
 			{{#if defaultValue}}
-				<input class='parameter required' minlength='1' name='{{name}}' placeholder='(required)' type='text' value='{{defaultValue}}'/>
+				<input data-threescale-name='{{threescale_name}}' class='parameter required' minlength='1' name='{{name}}' placeholder='(required)' type='text' value='{{defaultValue}}'/>
 			{{else}}
-				<input class='parameter required' minlength='1' name='{{name}}' placeholder='(required)' type='text' value=''/>
+				<input data-threescale-name='{{threescale_name}}' class='parameter required' minlength='1' name='{{name}}' placeholder='(required)' type='text' value=''/>
 			{{/if}}
 		{{/if}}
 	{{/if}}
