diff --git a/Cakefile b/Cakefile
index 281a4e7..7fc6490 100644
--- a/Cakefile
+++ b/Cakefile
@@ -1,5 +1,4 @@
 fs          = require 'fs'
-path        = require 'path'
 {exec}      = require 'child_process'
 less        = require 'less'
 
@@ -24,8 +23,8 @@ task 'clean', 'Removes distribution', ->
 
 task 'dist', 'Build a distribution', ->
   console.log "Build distribution in ./dist"
-  fs.mkdirSync('dist') if not path.existsSync('dist')
-  fs.mkdirSync('dist/lib') if not path.existsSync('dist/lib')
+  fs.mkdirSync('dist') if not fs.existsSync('dist')
+  fs.mkdirSync('dist/lib') if not fs.existsSync('dist/lib')
 
   appContents = new Array remaining = sourceFiles.length
   for file, index in sourceFiles then do (file, index) ->
